### PR TITLE
Add algosigner

### PR DESCRIPTION
**Pull request by MamoonAhmad**

### CODE REVIEW

File: examples/transit-react-basic/src/LoginPage.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  useEffect(() => {
+    const { getWalletProviders } = WAL.accessContext;
+    setState({ providers: getWalletProviders() });
+  }, []);
+
+  const login = async (wallet: WalletProvider) => {
+    WAL.accessContext.initWallet(wallet);
+    try {
+      const res = await wallet.connect("AlgoSignerApp");
```
Comment (traylewin) [inline examples/transit-react-basic/src/LoginPage.tsx:21]: Just to be clear, I would rename this app 'Demo Algo Signer App'

File: packages/eos-transit-algosigner-provider/src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        })
+
+        setTimeout(() => {
+          reject(`Cannot connect to "${shortName}" wallet provider`);
+        }, errorTimeout || 2500);
+      });
+    }
+
+    /** AlgoSigner doesn't store connection state hence no action required. */
+    function disconnect(): Promise<any> {
```
Comment (traylewin) [inline packages/eos-transit-algosigner-provider/src/plugin.ts:105]: return type should be typed. Please check for missing types everywhere. Shouldn't be using 'any' very often

File: packages/eos-transit-algosigner-provider/src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      return Promise.resolve(true);
+    }
+
+
+    /** Returns all accounts in a wallet. If network is provided in the constructor then it only returns accounts for that network.  */
+    async function discover(/* discoveryOptions: DiscoveryOptions */) {
```
Comment (traylewin) [inline packages/eos-transit-algosigner-provider/src/plugin.ts:111]: DiscoveryOptions shouldn't be commented out. Please review all code for other 'temporary hacks'

File: packages/eos-transit-algosigner-provider/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+}
+
+
+export interface AlgoSignerAccount{
+    account: string,
+    permission: string,
+    publicKey: string
+}
+
+
```
Comment (traylewin) [inline packages/eos-transit-algosigner-provider/src/types.ts:84]: Review code to clean up formatting and review all comments for accuracy, etc.

File: packages/eos-transit-algosigner-provider/src/plugin.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        signatures: [Buffer.from(signedTransaction.sig).toString("hex")], 
+        serializedTransaction: encodeToUint8Array({txn: signedTransaction.txn}),
+      };
+
+      return respone;
+    }
+  };
+}
+
+export interface AlgoSignerWalletProviderOptions {
```
Comment (traylewin) [inline packages/eos-transit-algosigner-provider/src/plugin.ts:57]: Shouldn't this be in the Types file?

**Review by traylewin — CHANGES_REQUESTED** [review]
Great work! Added comments - mostly for cleanup

File: packages/eos-transit-algosigner-provider/src/helper.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  export function processAccountForDiscovery({accounts, network, index}:{accounts: AlgoSignerAccountInfo[], network: AlgoNetworkType, index: number}): AlgoAccount[]{
+    return accounts.map(account => (
+        {
+            key: {
+                index: ++index,
+                key: toPublicKeyFromAddress(account.address),
+            },
+            note: JSON.stringify({
+                network,
+                account: account.address
```
Comment (traylewin) [inline packages/eos-transit-algosigner-provider/src/helper.ts:69]: {... account: walletAuth.accountName
 authorization: walletAuth.permission }

File: packages/eos-transit-algosigner-provider/src/helper.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+export function encodeToUint8Array(obj: any){
+    return msgpack.encode(obj, {sortKeys: true});
+
+}
+
+
+
+export function processAccount(account: AlgoSignerAccountInfo): WalletAuth{
```
Comment (traylewin) [inline packages/eos-transit-algosigner-provider/src/helper.ts:50]: rename processAccount() to getWalletAuthForAccount()

File: packages/eos-transit-algosigner-provider/src/helper.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+      accountName: account.address,
+      permission: ALGOSIGNER_DEFAULT_PERMISSION,
+      publicKey: toPublicKeyFromAddress(account.address)
+    }
+    
+  }
+  
+  export function processAccountForDiscovery({accounts, network, index}:{accounts: AlgoSignerAccountInfo[], network: AlgoNetworkType, index: number}): AlgoAccount[]{
+    return accounts.map(account => (
+        {
```
Comment (traylewin) [inline packages/eos-transit-algosigner-provider/src/helper.ts:62]: const walletAuth = getWalletAuthForAccount(account)

**Review by traylewin — CHANGES_REQUESTED** [review]

File: packages/eos-transit-algosigner-provider/src/plugin.ts [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+      return await (await _discover()).map(account => account.key.key);
+      
+    },
+
+    /** Signs a transaction using the private keys in AlgoSigner wallet. */
+    async sign({serializedTransaction}: SignatureProviderArgs): Promise<PushTransactionArgs>  {
+      
+      const txn = decodeUint8Array(serializedTransaction) as any;
+
```
Comment (traylewin) [inline packages/eos-transit-algosigner-provider/src/plugin.ts:36]: Remove fields from txn
flatFee, tag, name

**Review by traylewin — CHANGES_REQUESTED** [review]

File: packages/eos-transit-algosigner-provider/src/types.ts [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+    accountName: string;
+    permission: string;
+    publicKey: string;
+}
+
+export interface WalletProvider {
+    id: string;
+    meta?: WalletProviderMetadata;
+    signatureProvider: SignatureProvider;
+    connect(appName: string): Promise<any>;
```
[the remark this answers is not in the document]
Comment (MamoonAhmad) [reply packages/eos-transit-algosigner-provider/src/types.ts:209]: it returns a boolean to indicate that it is successfully connected.

**Pull request merged**